src/Network/Network/Models/PSSubnet.cs

```diff
@@ -27,6 +27,9 @@ public class PSSubnet : PSChildResource
         [JsonProperty(Order = 1)]
         public List<PSIPConfiguration> IpConfigurations { get; set; }
 
+        [JsonProperty(Order = 1)]
+        public PSNatGateway NatGateway { get; set; }
```

Comment:
Should this be
```
public PSResourceId NatGatewayId { get; set; }
```
to match specs?

Reply:
no, this should be similar to route table. We want to allow both id and resource